src/ol/webgl/TileTexture.js

```diff
@@ -36,8 +36,16 @@ function uploadImageTexture(gl, texture, image) {
  * @param {import("../DataTile.js").Data} data The pixel data.
  * @param {import("../size.js").Size} size The pixel size.
  * @param {number} bandCount The band count.
+ * @param {number} unpackAlignment The unpack alignment.
  */
-function uploadDataTexture(helper, texture, data, size, bandCount) {
+function uploadDataTexture(
+  helper,
+  texture,
+  data,
+  size,
+  bandCount,
+  unpackAlignment
```

Comment:
Instead of adding another argument, this function can determine the optimal unpack alignment based on `data.byteLength / size[1]`.  I've changed this in the latest commit.